### PR TITLE
Fix data-dir handling for crictl and helm charts

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

* Bump K3s version to pull through crictl.yaml fix from k3s-io/k3s#2475
* Update rancher/helm-controller to k3s-io/helm-controller to track with k3s (and update a couple docs to track the move as well)
* Pass through data-dir to helm charts as global.rke2DataDir so that templates can update host paths accordingly

#### Types of Changes ####

* K3s
* Helm chart bootstrap

#### Verification ####

* Start RKE2 with --data-dir in CLI flags or config file
* Note that /var/lib/rancher/rke2 does not exist
* Note that all pods start up successfully

#### Linked Issues ####

#474
#535
rancher/rke2-charts#36

### CODE REVIEW

File: pkg/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -370,10 +371,16 @@ func rewriteChart(fileName string, info os.FileInfo, systemDefaultRegistry strin
 
 	// Generally we should avoid using Set on HelmCharts since it cannot be overridden by HelmChartConfig,
 	// but in this case we need to do it in order to avoid potentially mangling the ValuesContent field by
-	// blindly appending content to it in order to set the global.systemDefaultRegistry value.
+	// blindly appending content to it in order to set values.
 	if chart.Spec.Set == nil {
 		chart.Spec.Set = map[string]intstr.IntOrString{}
 	}
+
+	if chart.Spec.Set["global.rke2DataDir"].StrVal != dataDir {
```
Comment (briandowns) [inline pkg/bootstrap/bootstrap.go:379]: What's the behavior here if the objects at that key doesn't exist and we try to call `Strval` on it?

File: pkg/bootstrap/bootstrap.go [hunk 2]
```diff
@@ -336,17 +337,17 @@ func setSystemDefaultRegistry(manifestsDir string, systemDefaultRegistry string)
...
 	}
 	return merr.NewErrors(errs...)
 }
 
-// rewriteChart applies systemDefaultRegistry settings to the file at fileName with associated info.
+// rewriteChart applies dataDir and systemDefaultRegistry settings to the file at fileName with associated info.
 // If the file cannot be decoded as a HelmChart, it is silently skipped. Any other IO error is considered
 // a failure.
-func rewriteChart(fileName string, info os.FileInfo, systemDefaultRegistry string, serializer *json.Serializer) error {
+func rewriteChart(fileName string, info os.FileInfo, dataDir string, systemDefaultRegistry string, serializer *json.Serializer) error {
```
Comment (briandowns) [inline pkg/bootstrap/bootstrap.go:350]:
```suggestion
func rewriteChart(fileName string, info os.FileInfo, dataDir, systemDefaultRegistry string, serializer *json.Serializer) error {
```

File: pkg/bootstrap/bootstrap.go (hunk 1 above)
Comment (brandond) [reply pkg/bootstrap/bootstrap.go:379]: `chart.Spec.Set` is typed as `map[string]intstr.IntOrString{}`, so if that's empty the map gives you back a default IntOrString struct. StrVal on the default struct is just an empty string.

Comment (briandowns) [reply pkg/bootstrap/bootstrap.go:379]: Cool. Just wanted to make sure we weren't introducing a point of panic.

**Review by briandowns — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]
LGTM

**Pull request merged**